### ISSUE TITLE
Update problem.

### ISSUE THREAD

**Issue by aerograf** [XoopsModules25x/tdmdownloads#74]
The first thing I encountered before starting the test is not a valid update, which led to the loss of the previous database (test). Could do something wrong, but then you need to write how to carry out the update.

**Comment by aerograf** [comment]
Upgrade from 1.66 to 2.0
XOOPS 2.5.10-Beta

**Comment by GregMage** [comment]
https://github.com/XoopsModules25x/tdmdownloads/blob/master/docs/Update.txt

**Comment by aerograf** [comment]
Yes, but the DB did not see and did not rename the database tables to the new format.
I'll try again.

**Comment by aerograf** [comment]
Module File for TDMDownloads Not Found!....
Unfortunately, it does not work. No matter how I did it. Basically writing an error with
Tdmdownloads \ Helper. If you start to change to TDMDownloads \ Helper, then the module stops working.
Installation on a clean, runs flawlessly.

**Comment by GregMage** [comment]
The problem: You have already updated from 1.66 to 2.0 (even if there was an error). It is necessary to delete the file tdmdownloads and to put back the files of the version 1.66 and to make an update of the module. The module is now in version 1.66. You follow the next procedure and all is well.

https://github.com/XoopsModules25x/tdmdownloads/blob/master/docs/Update.txt

**Comment by aerograf** [comment]
I'll try later what you offer.
There was a problem with "render ()". For some reason, display () works, and render () is not ... Because of this, the module in the admin panel does not display pages.

**Comment by cesagonchu** [comment]
A solution, while waiting to find better:
[https://www.frxoops.org/modules/newbb/viewtopic.php?post_id=242657#forumpost242657](url)

**Comment by mambax7** [comment]
Please ask babouille to fork the module and upload his changes. This will help to keep everything in one place :)

**Comment by mambax7** [comment]
And it's always good to look at the network diagram to see who has done what, and what are the latest versions, so we're aware of any overlaps:
https://github.com/XoopsModules25x/tdmdownloads/network

**Comment by cesagonchu** [comment]
Ok, but the only change he made was to rename tdmdownloads and everything else that calls the module to TDMDownloads.
Users who need to update the module can't because the module has been renamed to tdmdownloads (in lower case).

**Comment by mambax7** [comment]
They can update it by renaming the folder in the /modules on their server to "tdmdownloads"
It's for consistency how we use namespaces...

**Comment by cesagonchu** [comment]
babouille tried and this method did not work.
Do you have a tutorial?

**Comment by mambax7** [comment]
What version of TdmDownloads was he upgrading from? I could run the test here locally...

**Comment by cesagonchu** [comment]
We, 1.5 from 2010 and a user tried with 1.65

**Comment by cesagonchu** [comment]
We got "TDMDownloads not found" and the user got Error: Class 'TDMObjectTree' not found"

**Comment by mambax7** [comment]
A lot of the issues are related to the fact that "TDMDownloads" was hard coded in the module, so any change in in the module to "tdmdownloads" without also changing the name of the folder, will break it. I'll look into the issue and how to make the update easier.
